flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/table/catalog/impl/PulsarCatalogSupport.java

```diff
@@ -198,15 +199,24 @@ public CatalogTable getTable(ObjectPath tablePath) throws PulsarAdminException {
                         table.getComment(),
                         table.getPartitionKeys(),
                         fillDefaultOptionsFromCatalogOptions(table.getOptions()));
+            } catch (PulsarAdminException.NotFoundException e) {
+                throw e;
             } catch (Exception e) {
-                e.printStackTrace();
                 throw new CatalogException(
-                        "Failed to fetch metadata for explict table: " + tablePath.getObjectName());
+                        "Failed to fetch metadata for explict table: " + tablePath.getObjectName(),
+                        e);
             }
         } else {
             String existingTopic = findTopicForNativeTable(tablePath);
-            final SchemaInfo pulsarSchema = pulsarAdminTool.getPulsarSchema(existingTopic);
-            return schemaToCatalogTable(pulsarSchema, existingTopic);
+            try {
+                final SchemaInfo pulsarSchema = pulsarAdminTool.getPulsarSchema(existingTopic);
+                return schemaToCatalogTable(pulsarSchema, existingTopic);
+            } catch (PulsarAdminException.NotFoundException e) {
+                if (!tableExists(tablePath)) {
+                    throw e;
+                }
+                return schemaToCatalogTable(BytesSchema.BYTES.getSchemaInfo(), existingTopic);
```

Comment:
if this code is execute only at bootstrap I think that it is better that we log something and say that we fallback to the BYTES schema